File: 1317. Convert Integer to the Sum of Two No-Zero Integers/Solution.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        # let's break n in two parts: (x, n-x)
        # we will start from x == 1
        x = 1
        ans = []
        while True:
            # we will check if there is '0' in any of x and n-x
            # if yes we will update the value in the pair by updating x
            if '0' in str(n-x) or '0' in str(x):
                x += 1
            # else we will update the answer and will come out of the loop
            else:
                ans.append(x)
                ans.append(n-x)
                break
        return ans

File: 1317. Convert Integer to the Sum of Two No-Zero Integers/Solution.cpp
class Solution {
public:
    vector<int> getNoZeroIntegers(int n) {
        // intialize x as 1;
        int x = 1;
        vector<int> ans;
        while(true)
        {
            // convert (n-x) and x to string
            string a = to_string(n-x);
            string b = to_string(x);
        // if "0" is present in any index of a or b then we will increment the x and check the other pairs of (x) and (n-x).
            if(a.find("0")!= string::npos || b.find("0")!= string::npos)
            {
                x++;
            }
            // if no 0 is present in both number then we will push_back it in the vector.
            else{
                ans.push_back(x);
                ans.push_back(n-x);
                
                break;

            }
        }
        return ans;
        
    }
};
